### PR TITLE
Enable filtering entities by annotation values

### PR DESCRIPTION
**Pull request by gregorjerse**

### CODE REVIEW

**Review by dblenkus — COMMENTED** [review]
Add a test for this.

File: resolwe/flow/filters.py [hunk 1]
```diff
@@ -337,6 +339,63 @@ class EntityFilter(BaseCollectionFilter):
...
+            .values_list("type", flat=True)
+            .get()
+        )
+
+        if lookup_type not in allowed_lookups[field_type]:
+            raise ValueError(
+                f"Lookup '{lookup_type}' not supported for the field '{field_id}'."
+            )
+
+        post_processing = post_processing_map.get(field_type, lambda x: x)
```
Comment (dblenkus) [inline resolwe/flow/filters.py:385]: Use `None` as default and execute the post processing only when it is defined.

File: resolwe/flow/filters.py [hunk 2]
```diff
@@ -337,6 +339,63 @@ class EntityFilter(BaseCollectionFilter):
...
+        processing = lambda x: post_processing(
```
Comment (dblenkus) [inline resolwe/flow/filters.py:386]: Don't use generic variable names.

File: resolwe/flow/filters.py (hunk 1 above)
Comment (gregorjerse) [reply resolwe/flow/filters.py:385]: Fixed.

File: resolwe/flow/filters.py (hunk 2 above)
Comment (gregorjerse) [reply resolwe/flow/filters.py:386]: Fixed.

**Review by dblenkus — APPROVED** [review]

**Pull request merged**